pkgs/by-name/co/comfyui/package.nix

```diff
@@ -0,0 +1,151 @@
+{ lib
+, python311
+, linkFarm
+, writers
+, writeTextFile
+, fetchFromGitHub
+, stdenv
+, symlinkJoin
+, config
+, gpuBackend ? (
+  if config.cudaSupport
+  then "cuda"
+  else if config.rocmSupport
+  then "rocm"
+  else "none"
+)
+, modelsPath ? "/var/lib/comfyui/models"
+, inputPath ? "/var/lib/comfyui/input"
+, outputPath ? "/var/lib/comfyui/output"
+, tempPath ? "/var/lib/comfyui/temp"
+, userPath ? "/var/lib/comfyui/user"
+, customNodes ? []
+, models ? {
+    checkpoints = {};
+    clip = {};
+    clip_vision = {};
+    configs = {};
+    controlnet = {};
+    embeddings = {};
+    upscale_modules = {};
+    vae = {};
+    vae_approx = {};
+  }
+}:
+
+let
+
+  config-data = {
+    comfyui = {
+      base_path = modelsPath;
+      checkpoints = "${modelsPath}/checkpoints";
+      clip = "${modelsPath}/clip";
+      clip_vision = "${modelsPath}/clip_vision";
+      configs = "${modelsPath}/configs";
+      controlnet = "${modelsPath}/controlnet";
+      embeddings = "${modelsPath}/embeddings";
+      loras = "${modelsPath}/loras";
+      upscale_models= "${modelsPath}/upscale_models";
+      vae = "${modelsPath}/vae";
+      vae_approx = "${modelsPath}/vae_approx";
+    };
+  };
+
+  modelPathsFile = writeTextFile {
+    name = "extra_model_paths.yaml";
+    text = (lib.generators.toYAML {} config-data);
+  };
+
+  pythonEnv = (python311.withPackages (ps: with ps; [
+    pytorch-bin
+    # (
+    #   if gpuBackend == "cuda"
+    #   then torchWithCuda
+    #   else if gpuBackend == "rocm"
+    #   then torchWithRocm
+    #   else torch
+    # )
+    # torchsde
+    torchvision-bin
+    # torchaudio
+    (transformers.override { torch = pytorch-bin; })
+    (safetensors.override { torch = pytorch-bin; })
+    (accelerate.override { torch = pytorch-bin; })
+    # accelerate
+    torchsde
+    aiohttp
+    einops
+    kornia
+    pyyaml
+    pillow
+    scipy
+    psutil
+    tqdm
+  ] ++ (builtins.concatMap (node: node.dependencies) customNodes)));
+
+  executable = writers.writeDashBin "comfyui" ''
+    cd $out && \
+    ${pythonEnv}/bin/python comfyui \
```

Comment:
Was `comfyui` meant to be a module name (as when used with `-m ...`)? These are ideally done using pyproject.toml and `entrypoints`

Reply:
I admit I am new to the Python ecosystem.  I'll make it a point to take a look.